ios/sound.mm

```diff
@@ -24,209 +24,277 @@
-  }
-
-  // check device capabilities if it fulfills our requirements
-  const QString strStat = CheckDeviceCapabilities ( iDriverIdx );
-
-  // check if device is capable and if not the same device is used
-  if ( strStat.isEmpty() && ( strCurDevName.compare ( strDriverNames[iDriverIdx] ) != 0 ) )
-  {
-  }
-  // set the left/right in/output channels
-   
-  return strStat;
-   */
-    return "";
+    QMutexLocker locker ( &pSound->MutexAudioProcessCallback );
+    Float32*     pData = static_cast<Float32*> ( inInputData->mBuffers[0].mData );
+
+    // copy input data
+    for ( int i = 0; i < pSound->iCoreAudioBufferSizeMono; i++ )
+    {
+        // copy left and right channels separately
+        pSound->vecsTmpAudioSndCrdStereo[2 * i]     = (short) ( pData[2 * i] * _MAXSHORT );     // left
+        pSound->vecsTmpAudioSndCrdStereo[2 * i + 1] = (short) ( pData[2 * i + 1] * _MAXSHORT ); // right
+    }
+    pSound->ProcessCallback ( pSound->vecsTmpAudioSndCrdStereo );
 }
 
-QString CSound::CheckDeviceCapabilities ( const int iDriverIdx )
+/* Implementation *************************************************************/
+CSound::CSound ( void ( *fpNewProcessCallback ) ( CVector<short>& psData, void* arg ),
+                 void*          arg,
+                 const QString& strMIDISetup,
+                 const bool,
+                 const QString& ) :
+    CSoundBase ( "CoreAudio iOS", fpNewProcessCallback, arg, strMIDISetup )
 {
-  /*// This function checks if our required input/output channel
-  // properties are supported by the selected device. If the return
-  // string is empty, the device can be used, otherwise the error
-  // message is returned.
-
-  // check sample rate of input
-  audioSession.setPreferredSampleRate( SYSTEM_SAMPLE_RATE_HZ );
-  // if false, try to set it
-  if ( audioSession.sampleRate != SYSTEM_SAMPLE_RATE_HZ )
-  {
-    throw CGenErr ( tr ("Could not set sample rate. Please close other applications playing audio.") );
-  }
-
-  // special case with 4 input channels: support adding channels
-    if ( ( lNumInChan == 4 ) && bInputChMixingSupported )
+    try
     {
-        // add four mixed channels (i.e. 4 normal, 4 mixed channels)
-        lNumInChanPlusAddChan = 8;
-
-        for ( int iCh = 0; iCh < lNumInChanPlusAddChan; iCh++ )
-        {
-            int iSelCH, iSelAddCH;
-
-            GetSelCHAndAddCH ( iCh, lNumInChan, iSelCH, iSelAddCH );
-
-            if ( iSelAddCH >= 0 )
-            {
-                // for mixed channels, show both audio channel names to be mixed
-                channelInputName[iCh] =
-                    channelInputName[iSelCH] + " + " + channelInputName[iSelAddCH];
-            }
-        }
+        NSError* audioSessionError = nil;
+
+        [[AVAudioSession sharedInstance] setCategory:AVAudioSessionCategoryPlayAndRecord error:&audioSessionError];
+        [[AVAudioSession sharedInstance] requestRecordPermission:^( BOOL granted ) {
```

Comment:
Indentation appears to be off here.  It should indent four spaces (at least), it's only indented two.  The block type is `[{..}]` rather than `{..}`, which is presumably confusing it.

Reply:
ident 4 actually

Reply:
I meant following the `{` at the end of the line - i.e. what's currently line 109 is only two spaces indented from line 108, whereas it should be 4.

Reply:
I see now!